guides/hack/21-XHP/19-migration.md

```diff
@@ -31,11 +31,12 @@ strings around in the end.
 You could make `render_component` into a class:
 
 ```Hack
-// Assume class Uri
-class :ui:component-link extends :x:element {
-  attribute Uri $uri @required;
+namespace ui;
+
+class link extends x\element {
+  attribute Uri $uri @required;  // Assume class Uri
   attribute string $text @required;
```

Comment:
TIL that you can have more than one attributes declaration in a class.
I thought that you had to put all attributes in one declaration.

Reply:
I have found the source of my confusion. In the [hhvm 4.3.0 blog post](https://hhvm.com/blog/2019/04/30/hhvm-4.3.0.html) the rules for multiple `children` and `category` declarations changed. Multiple `attribute` declarations are still allowed.